packages/sanity/src/core/form/studio/tree-editing/components/TreeEditingMenu.tsx

```diff
@@ -0,0 +1,262 @@
+    padding: 2px;
+    padding-right: 3px;
+    box-sizing: border-box;
+    transition: inherit;
+
+    &[data-selected='true'] {
+      background-color: ${selectedBg};
+      border-radius: ${theme.sanity.radius[2]}px;
+    }
+
+    [data-ui='ExpandButton'],
+    [data-ui='NavigateButton'] {
+      transition: inherit;
+      background-color: inherit;
+    }
+
+    @media (hover: hover) {
+      &:hover {
+        &[data-selected='false'] {
+          background-color: ${defaultHoverBg};
+          border-radius: ${getTheme_v2(theme).radius[2]}px;
+
+          [data-ui='ExpandButton']:hover {
+            background-color: ${selectedHoverBg};
+          }
+        }
+
+        [data-ui='ExpandButton']:hover {
+          background-color: ${defaultHoverBg};
+        }
+      }
+    }
+  `
+})
+
+interface TreeEditingMenuItemProps {
+  item: TreeEditingMenuItem
+  onPathSelect: (path: Path) => void
+  selectedPath: Path | null
+  siblingHasChildren?: boolean
+}
+
+function MenuItem(props: TreeEditingMenuItemProps) {
+  const {item, onPathSelect, selectedPath, siblingHasChildren} = props
+  const {children} = item
+  const hasChildren = children && children.length > 0
+  const [open, setOpen] = useState<boolean>(false)
+  const {t} = useTranslation()
+
+  const [rootElement, setRootElement] = useState<HTMLElement | null>(null)
+
+  const selected = useMemo(() => isEqual(item.path, selectedPath), [item.path, selectedPath])
+  const isArrayParent = useMemo(() => !isArrayItemPath(item.path), [item.path])
+
+  const {value} = useValuePreviewWithFallback({
+    schemaType: item.schemaType,
+    value: item.value,
+  })
+
+  const title = useMemo(() => {
+    // If the item is an array parent, we want to show the schema type title
+    if (isArrayParent) {
+      return getSchemaTypeTitle(item.schemaType)
+    }
+
+    // Else, we show the preview title
+    return value.title
+  }, [isArrayParent, item.schemaType, value.title])
+
+  const handleClick = useCallback(() => {
+    onPathSelect(item.path)
+  }, [item.path, onPathSelect])
+
+  const handleExpandClick = useCallback(() => setOpen((v) => !v), [])
+
+  const icon = useMemo(() => {
+    if (!hasChildren) return null
+
+    return <AnimateChevronIcon data-expanded={open ? 'true' : 'false'} />
+  }, [hasChildren, open])
+
+  useEffect(() => {
+    const hasOpen = hasOpenChild(item, selectedPath)
+
+    if (hasOpen) {
+      setOpen(true)
+    }
+  }, [item, selectedPath])
+
+  // Scroll to the selected item
+  useEffect(() => {
+    if (!rootElement || !selected) return
+
+    scrollIntoViewIfNeeded(rootElement, SCROLL_BEHAVIOR_OPTIONS)
+  }, [rootElement, selected])
+
+  return (
+    <Stack
+      aria-expanded={open}
+      as="li"
+      key={toString(item.path)}
+      ref={setRootElement}
+      role="treeitem"
+      space={1}
+    >
+      <Card data-as="button" radius={2} tone="inherit">
+        <ItemFlex align="center" data-selected={selected} data-testid="side-menu-item">
```

Comment:
The test ID should be renamed to `tree-editing-sidebar-item` to align with the naming of other test IDs, i.e., `tree-editing-<target>`.